contracts/CollateralPool.sol

```diff
@@ -77,6 +83,14 @@ contract CollateralPool is PiAdmin, Pausable {
         piGlobal = _piGlobal;
     }
 
+    /**
+     * @dev Modifier to restrict access to only EOA accounts. Only applies if `onlyEOA` is true.
+     */
+    modifier maybeOnlyEOA() {
+        if (onlyEOA && msg.sender != tx.origin) revert OnlyEOA();
```

Comment:
let's check for bytecode better =) like here: 
https://ethereum.stackexchange.com/questions/15641/how-does-a-contract-find-out-if-another-address-is-a-contract